src/transformers/models/mt5/modeling_mt5.py

```diff
@@ -566,8 +566,12 @@ def forward(
         attention_outputs = self_attention_outputs[2:]  # Keep self-attention outputs and relative position weights
 
         # clamp inf values to enable fp16 training
-        if hidden_states.dtype == torch.float16 and torch.isinf(hidden_states).any():
-            clamp_value = torch.finfo(hidden_states.dtype).max - 1000
+        if hidden_states.dtype == torch.float16:
+            clamp_value = torch.where(
+                torch.isinf(hidden_states).any(),
+                torch.finfo(hidden_states.dtype).max - 1000,
+                torch.finfo(hidden_states.dtype).max,
+            )
```

Comment:
Just out of curiosity, why clamp infs to max - 1000, but float values to max?

Reply:
In the original scenario, we only clamp when there is an inf present. When there is no inf, no clamping is done. This performs the same operation with slightly different logic where if we do not see an inf it clamps the values to the maximum range of float (ie. no actual clamping is performed).